app.py

```diff
@@ -19,11 +24,21 @@ def root():
 @app.route("/upload", methods=["GET", "POST"])
 def upload():
     if request.method == "POST":
+        uploads = {"success": [], "failed": []}
         for file_ in request.files.getlist("file"):
             filename = secure_filename(file_.filename)
-            file_.save(os.path.join(app.config["UPLOAD_FOLDER"], filename))
-
-        return Response("success")
+            with io.BytesIO() as stream:
```

Comment:
would you benefit from a function, perhaps, `ingest_file` that does the streaming and calls within it the identification (and other future steps) and return Success of Fail with a reason. I am not very clear but a lot is happening in this upload function right now and it will probably only grow to spill into run_pipeline

Reply:
this was a _great_ idea! doing this allowed us to test the function and, incidentally, discover it didn't actually work as originally written 😅 

tested and fixed now. tests are amazing 😄